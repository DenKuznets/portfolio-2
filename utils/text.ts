const text = {
    showmore: 'Подробнее',
    backButton: 'Назад',
    myWorks: 'Мои проекты',
    allWorks: 'Все проекты',
    examples: 'Примеры работ',
    aboutMe: 'Обо мне',
    links: {
        inst: 'https://www.instagram.com/denis_kyznecov/',
        git: 'https://github.com/DenKuznets',
        tg: 'https://t.me/DenKuznets',
        project: '/projects/'
    },
    hero: {
        fname: 'Денис',
        lname: 'Кузнецов',
        profession: 'front-end разработчик',
        aboutme: 'верстаю сайты на react и styled-components',
        stack: 'СТЭК: react, nextjs, ts, mui, tailwind, styled-components, redux, rest, graphql, jest, react-testing-library, storybook',
        contactme: 'Связаться',
        socials: 'я в соц сетях'
    }
};

export default text;
