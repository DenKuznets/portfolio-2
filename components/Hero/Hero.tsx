import Image from 'next/image';
import text from '../../utils/text';
import instIcon from '../../public/images/instagram-logo.svg';
import gitIcon from '../../public/images/github-logo.svg';
import avatarPic from '../../public/images/avatar-violet-bg.webp';
import MyLink from '../MyLink/MyLink';
import Link from 'next/link';

export const hero_testids = {
    hero_container: 'hero_container'
};

const Hero = () => {
    const socials = [
        { link: text.links.git, icon: gitIcon },
        { link: text.links.inst, icon: instIcon }
    ];
    return (
        <div
            data-testid={hero_testids.hero_container}
            className="container md:mt-[90px] mx-auto flex flex-col items-center lg:grid lg:grid-cols-2 lg:justify-between lg:gap-8 uhd:max-w-[1920px]"
        >
            <div>
                <h1 className="font-bold text-5xl sm:text-6xl">
                    {text.hero.fname} {text.hero.lname}
                </h1>
                <div className="mt-1 flex items-center justify-end md:justify-start">
                    <hr className="mr-4 w-12 border-black md:w-20" />
                    <span className="first-letter:capitalize md:text-2xl">
                        {text.hero.profession}
                    </span>
                </div>
                <p className="color-black-900 mt-10 text-center md:text-start  uppercase leading-normal">
                    {text.hero.stack}
                </p>

                <div className="mt-14 flex flex-col items-center gap-4 md:flex-row ">
                    <MyLink violet href={text.links.tg}>
                        {text.hero.contactme}
                    </MyLink>
                    <MyLink className='whitespace-nowrap' href="/allworks">{text.myWorks}</MyLink>
                </div>
                <div className="flex justify-center items-center mt-14 md:justify-start">
                    <span className="first-letter:capitalize">
                        {text.hero.socials}:
                    </span>
                    <div className="flex px-4 gap-6">
                        {socials.map((obj) => (
                            <Link
                                className="hover:scale-125 transition-all"
                                key={obj.link}
                                href={obj.link}
                            >
                                <Image
                                    width={25}
                                    height={25}
                                    src={obj.icon}
                                    alt={obj.link}
                                />
                            </Link>
                        ))}
                    </div>
                </div>
            </div>
            <Image
                className="max-lg:hidden rounded-[30%_70%_70%_30%_/_30%_30%_70%_70%] shadow-[15px_15px_50px_rgba(0,0,0,0.2)]"
                width={700}
                height={900}
                src={avatarPic}
                alt="man in yellow jacket"
            />
        </div>
    );
};

export default Hero;
