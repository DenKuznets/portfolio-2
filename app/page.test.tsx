import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Home from './page';
import { hero_testids } from '@/components/Hero/Hero';
import text from '@/utils/text';

test('renders correctly', () => {
    render(<Home />);
    const header = screen.getByTestId('header');
    const hero = screen.getByTestId(hero_testids.hero_container);
    const examplesh2 = screen.getByRole('heading', { level: 2 });
    const sections = screen.getAllByTestId('section');
    const allProjectsLink = screen.getByRole('link', {
        name: text.allWorks
    });

    expect(header).toBeInTheDocument();
    expect(hero).toBeInTheDocument();
    expect(examplesh2).toHaveTextContent(text.examples);
    expect(sections).toHaveLength(3);
    expect(allProjectsLink).toHaveAttribute('href', '/projects');
});

test('renders Home unchanged', () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
});
