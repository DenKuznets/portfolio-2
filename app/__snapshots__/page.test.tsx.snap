// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home unchanged 1`] = `
<div>
  <main
    class=" mx-auto flex flex-col uhd:max-w-[1920px]"
  >
    <div
      class="container sm:mt-[40px] md:mt-[60px] mx-auto flex flex-col items-center lg:grid lg:grid-cols-2 lg:justify-between lg:gap-8 px-2 md:px-4"
      data-testid="hero_container"
      id="#hero"
    >
      <div>
        <h1
          class="font-bold text-5xl sm:text-6xl"
        >
          Денис
           
          Кузнецов
        </h1>
        <div
          class="mt-1 flex items-center justify-end md:justify-start"
        >
          <hr
            class="mr-4 w-12 border-black md:w-20"
          />
          <span
            class="first-letter:capitalize md:text-2xl"
          >
            front-end разработчик
          </span>
        </div>
        <p
          class="color-black-900 mt-10 text-center md:text-start  uppercase leading-normal"
        >
          СТЭК: react, nextjs, ts, mui, tailwind, styled-components, redux, rest, graphql, jest, react-testing-library, storybook
        </p>
        <div
          class="mt-14 flex flex-col items-center gap-4 md:flex-row "
        >
          <a
            class="hover:outline-[currentColor] text-inherit hover:outline-1 hover:outline outline-none text-sm sm:text-lg border-none transition-all flex w-fit items-center rounded-2xl px-8 sm:px-10 py-4 sm:py-5 [&>svg]:hover:rotate-45 [&>svg]:ml-2  [&>svg]:transition-all 
            bg-violet [&&]:text-white [&&]:hover:text-amber-400 [&>svg]:hover:fill-amber-400 [&&]:hover:outline-none"
            href="https://t.me/DenKuznets"
          >
            Связаться
            <svg
              fill="white"
              height="30px"
              style="fill: currentColor;"
              viewBox="0 0 24 24"
              width="30px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                data-name="Location Arrow 1"
                id="Location_Arrow_1"
              >
                <path
                  d="M14.472,20.937a1.438,1.438,0,0,1-1.3-.812L10.3,14.343a1.418,1.418,0,0,0-.642-.641L3.874,10.831A1.462,1.462,0,0,1,4.06,8.136l14.952-5a1.46,1.46,0,0,1,1.849,1.847l-5,14.952a1.439,1.439,0,0,1-1.284.994C14.543,20.936,14.507,20.937,14.472,20.937ZM19.479,4.063a.488.488,0,0,0-.149.024h0l-14.952,5a.46.46,0,0,0-.058.849L10.1,12.805A2.444,2.444,0,0,1,11.2,13.9l2.87,5.782a.443.443,0,0,0,.445.255.45.45,0,0,0,.4-.312l5-14.953a.462.462,0,0,0-.433-.607Z"
                />
              </g>
            </svg>
          </a>
          <a
            class="hover:outline-[currentColor] text-inherit hover:outline-1 hover:outline outline-none text-sm sm:text-lg border-none transition-all flex w-fit items-center rounded-2xl px-8 sm:px-10 py-4 sm:py-5 [&>svg]:hover:rotate-45 [&>svg]:ml-2  [&>svg]:transition-all 
            whitespace-nowrap text-black"
            href="/projects"
          >
            Мои проекты
            <svg
              class="fill-current"
              height="30px"
              style="stroke: currentColor;"
              viewBox="0 0 24 24"
              width="30px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7 17L17 7M17 7H8M17 7V16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </a>
        </div>
        <div
          class="flex justify-center items-center mt-8 md:mt-14 md:justify-start"
        >
          <span
            class="first-letter:capitalize text-sm sm:text-lg"
          >
            я в соц сетях
            :
          </span>
          <div
            class="flex px-4 gap-6"
          >
            <a
              class="hover:scale-125 transition-all"
              href="https://github.com/DenKuznets"
            >
              <img
                alt="https://github.com/DenKuznets"
                data-nimg="1"
                decoding="async"
                height="25"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
                style="color: transparent;"
                width="25"
              />
            </a>
            <a
              class="hover:scale-125 transition-all"
              href="https://www.instagram.com/denis_kyznecov/"
            >
              <img
                alt="https://www.instagram.com/denis_kyznecov/"
                data-nimg="1"
                decoding="async"
                height="25"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
                style="color: transparent;"
                width="25"
              />
            </a>
          </div>
        </div>
      </div>
      <img
        alt="man in yellow jacket"
        class="max-lg:hidden rounded-[30%_70%_70%_30%_/_30%_30%_70%_70%] shadow-[15px_15px_50px_rgba(0,0,0,0.2)]"
        data-nimg="1"
        decoding="async"
        height="900"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=1920&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=750&q=75 1x, /_next/image?url=%2Fimg.jpg&w=1920&q=75 2x"
        style="color: transparent;"
        width="700"
      />
    </div>
    <h2
      class="container mt-4 bg-violet py-10 text-center text-3xl font-bold text-white underline sm:mt-8 sm:text-4xl"
      id="examples"
    >
      Примеры работ
    </h2>
    <div
      class="flex-col flex gap-8"
    >
      <div
        class="container px-2 md:px-4 lg:grid grid-cols-2 items-center lg:gap-8 py-8 justify-between bg-violet text-white"
        data-testid="section"
      >
        <div>
          <h3
            class="font-bold leading-normal mb-4 first-letter:capitalize"
          >
            rock paper scissors
          </h3>
          <ul
            class="flex flex-wrap gap-3 items-center mb-4"
          >
            <li
              class="whitespace-nowrap px-4 py-2 border-dotted rounded-md border-[currentColor] border text-sm sm:text-lg"
            >
              REACT
            </li>
            <li
              class="whitespace-nowrap px-4 py-2 border-dotted rounded-md border-[currentColor] border text-sm sm:text-lg"
            >
              MUI
            </li>
            <li
              class="whitespace-nowrap px-4 py-2 border-dotted rounded-md border-[currentColor] border text-sm sm:text-lg"
            >
              TS
            </li>
            <li
              class="whitespace-nowrap px-4 py-2 border-dotted rounded-md border-[currentColor] border text-sm sm:text-lg"
            >
              STORYBOOK
            </li>
            <li
              class="whitespace-nowrap px-4 py-2 border-dotted rounded-md border-[currentColor] border text-sm sm:text-lg"
            >
              UNIT TESTING
            </li>
          </ul>
          <a
            class="text-sm sm:text-lg"
            href="https://github.com/DenKuznets/rock-paper-scissors"
          >
            Github
          </a>
           
          |
           
          <a
            class="text-sm sm:text-lg"
            href="https://rock-paper-scissors-kappa-tawny.vercel.app/"
          >
            Demo
          </a>
          <a
            class="hover:outline-[currentColor] text-inherit hover:outline-1 hover:outline outline-none text-sm sm:text-lg border-none transition-all flex w-fit items-center rounded-2xl px-8 sm:px-10 py-4 sm:py-5 [&>svg]:hover:rotate-45 [&>svg]:ml-2  [&>svg]:transition-all 
            mt-4 mb-4 lg:mb-0 hover:outline-inherit"
            href="/projects/1"
          >
            Подробнее
            <svg
              class="fill-current"
              height="30px"
              style="stroke: currentColor;"
              viewBox="0 0 24 24"
              width="30px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7 17L17 7M17 7H8M17 7V16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </a>
        </div>
        <a
          class="relative mt-8 lg:mt-0"
          href="/projects/1"
        >
          <img
            alt="rock paper scissors project image"
            class="filter-none shadow-2xl hover:saturate-150 transition-all duration-300"
            data-nimg="1"
            decoding="async"
            height="600"
            loading="lazy"
            src="/_next/image?url=%2Fimages%2Fworks-preview%2Frpc.webp&w=1920&q=75"
            srcset="/_next/image?url=%2Fimages%2Fworks-preview%2Frpc.webp&w=828&q=75 1x, /_next/image?url=%2Fimages%2Fworks-preview%2Frpc.webp&w=1920&q=75 2x"
            style="color: transparent;"
            width="800"
          />
        </a>
      </div>
      <div
        class="container px-2 md:px-4 lg:grid grid-cols-2 items-center lg:gap-8 py-8 justify-between "
        data-testid="section"
      >
        <div>
          <h3
            class="font-bold leading-normal mb-4 first-letter:capitalize"
          >
            коммерческий сайт
          </h3>
          <ul
            class="flex flex-wrap gap-3 items-center mb-4"
          >
            <li
              class="whitespace-nowrap px-4 py-2 border-dotted rounded-md border-[currentColor] border text-sm sm:text-lg"
            >
              HTML CSS JS
            </li>
          </ul>
          <a
            class="text-sm sm:text-lg"
            href="https://github.com/DenKuznets/tes.loc"
          >
            Github
          </a>
           
          |
           
          <a
            class="text-sm sm:text-lg"
            href="https://tul-energo-service.ru/"
          >
            Demo
          </a>
          <a
            class="hover:outline-[currentColor] text-inherit hover:outline-1 hover:outline outline-none text-sm sm:text-lg border-none transition-all flex w-fit items-center rounded-2xl px-8 sm:px-10 py-4 sm:py-5 [&>svg]:hover:rotate-45 [&>svg]:ml-2  [&>svg]:transition-all 
            mt-4 mb-4 lg:mb-0 hover:outline-inherit"
            href="/projects/2"
          >
            Подробнее
            <svg
              class="fill-current"
              height="30px"
              style="stroke: currentColor;"
              viewBox="0 0 24 24"
              width="30px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7 17L17 7M17 7H8M17 7V16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </a>
        </div>
        <a
          class="relative mt-8 lg:mt-0"
          href="/projects/2"
        >
          <img
            alt="коммерческий сайт project image"
            class="filter-none shadow-2xl hover:saturate-150 transition-all duration-300"
            data-nimg="1"
            decoding="async"
            height="600"
            loading="lazy"
            src="/_next/image?url=%2Fimages%2Fworks-preview%2Ftes.webp&w=1920&q=75"
            srcset="/_next/image?url=%2Fimages%2Fworks-preview%2Ftes.webp&w=828&q=75 1x, /_next/image?url=%2Fimages%2Fworks-preview%2Ftes.webp&w=1920&q=75 2x"
            style="color: transparent;"
            width="800"
          />
        </a>
      </div>
      <div
        class="container px-2 md:px-4 lg:grid grid-cols-2 items-center lg:gap-8 py-8 justify-between bg-customYellow"
        data-testid="section"
      >
        <div>
          <h3
            class="font-bold leading-normal mb-4 first-letter:capitalize"
          >
            тестовое задание
          </h3>
          <ul
            class="flex flex-wrap gap-3 items-center mb-4"
          >
            <li
              class="whitespace-nowrap px-4 py-2 border-dotted rounded-md border-[currentColor] border text-sm sm:text-lg"
            >
              REACT
            </li>
            <li
              class="whitespace-nowrap px-4 py-2 border-dotted rounded-md border-[currentColor] border text-sm sm:text-lg"
            >
              STYLED-COMPONENTS
            </li>
            <li
              class="whitespace-nowrap px-4 py-2 border-dotted rounded-md border-[currentColor] border text-sm sm:text-lg"
            >
              REST
            </li>
          </ul>
          <a
            class="text-sm sm:text-lg"
            href="https://github.com/DenKuznets/dls-test"
          >
            Github
          </a>
           
          |
           
          <a
            class="text-sm sm:text-lg"
            href="https://dls-test.vercel.app/"
          >
            Demo
          </a>
          <a
            class="hover:outline-[currentColor] text-inherit hover:outline-1 hover:outline outline-none text-sm sm:text-lg border-none transition-all flex w-fit items-center rounded-2xl px-8 sm:px-10 py-4 sm:py-5 [&>svg]:hover:rotate-45 [&>svg]:ml-2  [&>svg]:transition-all 
            mt-4 mb-4 lg:mb-0 hover:outline-inherit"
            href="/projects/3"
          >
            Подробнее
            <svg
              class="fill-current"
              height="30px"
              style="stroke: currentColor;"
              viewBox="0 0 24 24"
              width="30px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7 17L17 7M17 7H8M17 7V16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </a>
        </div>
        <a
          class="relative mt-8 lg:mt-0"
          href="/projects/3"
        >
          <img
            alt="тестовое задание project image"
            class="filter-none shadow-2xl hover:saturate-150 transition-all duration-300"
            data-nimg="1"
            decoding="async"
            height="600"
            loading="lazy"
            src="/_next/image?url=%2Fimages%2Fworks-preview%2Ftest1.webp&w=1920&q=75"
            srcset="/_next/image?url=%2Fimages%2Fworks-preview%2Ftest1.webp&w=828&q=75 1x, /_next/image?url=%2Fimages%2Fworks-preview%2Ftest1.webp&w=1920&q=75 2x"
            style="color: transparent;"
            width="800"
          />
        </a>
      </div>
    </div>
    <a
      class="hover:outline-[currentColor] text-inherit hover:outline-1 hover:outline outline-none text-sm sm:text-lg border-none transition-all flex w-fit items-center rounded-2xl px-8 sm:px-10 py-4 sm:py-5 [&>svg]:hover:rotate-45 [&>svg]:ml-2  [&>svg]:transition-all 
            mx-auto mt-8 font-bold"
      href="/projects"
    >
      Все проекты
      <svg
        class="fill-current"
        height="30px"
        style="stroke: currentColor;"
        viewBox="0 0 24 24"
        width="30px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 17L17 7M17 7H8M17 7V16"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </a>
  </main>
</div>
`;
